servlet/src/main/java/io/undertow/servlet/handlers/security/SSLInformationAssociationHandler.java

```diff
@@ -94,25 +96,30 @@ public static int getKeyLength(String cipherSuite) {
      *         negotiate the SSL connection. <br>
      *         Will be null if the chain is missing or empty.
      */
-    private X509Certificate[] getCerts(SSLSessionInfo session) {
-        try {
-            javax.security.cert.X509Certificate[] javaxCerts = session.getPeerCertificateChain();
-            if (javaxCerts == null || javaxCerts.length == 0) {
-                return null;
-            }
-            X509Certificate[] javaCerts = new X509Certificate[javaxCerts.length];
-            java.security.cert.CertificateFactory cf = java.security.cert.CertificateFactory.getInstance("X.509");
-            for (int i = 0; i < javaxCerts.length; i++) {
-                byte[] bytes = javaxCerts[i].getEncoded();
-                ByteArrayInputStream stream = new ByteArrayInputStream(bytes);
-                javaCerts[i] = (X509Certificate) cf.generateCertificate(stream);
-            }
-
-            return javaCerts;
-        } catch (Exception e) {
-            return null;
-        }
-                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        }
+  private X509Certificate[] getCerts(SSLSessionInfo session) {
+    try {
+      return session.getPeerCertificateChain();
+    } catch (SSLPeerUnverifiedException | RenegotiationRequiredException e) {
+      return null;
+    }
+//        try {
+//            javax.security.cert.X509Certificate[] javaxCerts = session.getPeerCertificateChain();
+//            if (javaxCerts == null || javaxCerts.length == 0) {
+//                return null;
+//            }
+//            X509Certificate[] javaCerts = new X509Certificate[javaxCerts.length];
+//            java.security.cert.CertificateFactory cf = java.security.cert.CertificateFactory.getInstance("X.509");
+//            for (int i = 0; i < javaxCerts.length; i++) {
+//                byte[] bytes = javaxCerts[i].getEncoded();
+//                ByteArrayInputStream stream = new ByteArrayInputStream(bytes);
+//                javaCerts[i] = (X509Certificate) cf.generateCertificate(stream);
+//            }
+//
+//            return javaCerts;
+//        } catch (Exception e) {
+//            return null;
+//        }
```

Comment:
here as well

Reply:
Fixed.

Reply:
I can't find where this is fixed?

Reply:
Same here, now I can find where it is fixed! The < and > navigation of the PR review screen did not show those commits :-)